### PR TITLE
PY-Chang turns in proposal [attempt 1]

### PR DESCRIPTION
**Pull request by PY-Chang**
---
A good PR should include the following parts:

- [x] A clear title
- [x] A meaningful message for PR, as well as its commits
- [x] from your specific branch (***not master***) merging to teacher's `HW` branch (***not master***, again)
- [x] excluding any irrelevant files, such as binaries, text files, or dot files
- [ ] passing tests/CI

### CODE REVIEW

File: project/PY-Chang/README.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
+# YAMM
+Yet Another Matrix Multiplier!
+
+## Basic Information
+> Github repository: https://github.com/PY-Chang/YAMM.git
```
Comment (yungyuc) [inline project/PY-Chang/README.md]: It's OK to use GPL, but if it's OK, I prefer BSD or MIT more.  Using GPL prevents commercial companies from using your code.

File: project/PY-Chang/README.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+This is a matrix multiplier based on C++ and Python hybrid system. This project implements the classic Strassen algorithm with the help of parallel computing to achieve better performance on modern computer.
+
+## Problem to Solve and Prospective Users
+Matrix multiplication is one of the most important linear algebra operation in the modern computing world. For decades, many researches have been invested to improve the efficiency of matrix multiplication. 
+
+In 1969, Volker Strassen developed an algorithm, namely, Strassen algorithm which redeuces the time complicity of the matrix multiplication down to around O(n<sup>2.8074</sup>) in the first time. 
+
+
+Although in recent years researchers have developed faster algorithms than Strassen's, those algorithms have been viewed as galactic algorithms which means non-practical to present-day computers.
```
Comment (yungyuc) [inline project/PY-Chang/README.md]: This is not true.  Advanced algorithms are used in commercial tools already.  They are, oftentimes, not worth the time for being implemented in code for researches in fields other than linear algebra itself.

File: project/PY-Chang/README.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+Therefore in this project, we will implement the Strassen algorithm based on C++ and provide an easy-to-use python API to the user who is not familiar with the algorithm and parallel computing.
```
Comment (yungyuc) [inline project/PY-Chang/README.md]: Merely implementing the Strassen algorithm will be too small a scope for a term project.  It usually constitutes one homework assignment in a numerical-analysis or linear-algebra course.  And you will not be able to come up with an API that is more useful than LAPACK.  There will be little to say in your presenetation.

It is a good direction to do something with linear algebra, though.  I suggest you to cover more  work in linear algebra.

File: project/PY-Chang/README.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+## System Architecture
+The system would provide a class named `Matrix` to handle input and output. Also, for the sake of convenience, the system would be designed to take NumPy array as input and output as well. Once the input matrixes are provided, the system would use the Strassen algorithm to calculate the result of the matrix multiplication.
+
+## API Description
+#### Basic API
+* `Matrix()`: declare a `Matrix` instance
```
Comment (yungyuc) [inline project/PY-Chang/README.md]: Please provide an analysis for the coverage of BLAS API.

File: project/PY-Chang/README.md [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+3. Since the API is build on python, we'll use `pytest` to do unit test.
+4. The documentation will be provided as `README.md` on the GitHub repository. 
+
+## Schedule
+* Week 1 (4/4): Building the class `Matrix`.
+* Week 2 (4/11): Implementing the Strassen algorithm.
+* Week 3 (4/18): Improving performance by adding parallel computing.(1/2)
+* Week 4 (4/25):  Improving performance by adding parallel computing.(2/2)
+* Week 5 (5/2): Using `pybind11` to wrap into Python.
+* Week 6 (5/9): Building python unit test to test the correctness of the calculation.
```
Comment (yungyuc) [inline project/PY-Chang/README.md]: Doing testing in the 6th weeks leaves you no time to correct mistakes you make.

**Review by yungyuc — CHANGES_REQUESTED** [review]
Please add an entry pointing to your proposal at the bottom of the `README.rst` in the upper-level directory.

It is a good direction to do something with linear algebra.  I suggest you to cover more  work in linear algebra.

File: project/PY-Chang/README.md (hunk 3 above)
Comment (PY-Chang) [reply project/PY-Chang/README.md]: @yungyuc What if I implement part of BLAS single precision routines and wrap them into python? The purpose is to let users use the same function prototype to call the API in the python as in c. (Although I know there has been CBLAS already...)

Comment (yungyuc) [reply project/PY-Chang/README.md]: If you plan to reimplement BLAS, you need to justify with the value of your work.  It does not need to be fancy, but you can reference some BLAS (re)implementations that are done over the years.

**Comment by yungyuc** [comment]
@PY-Chang It is already overdue.  Please finalize the PR as soon as possible.

**Review by yungyuc — APPROVED** [review]

**Pull request merged**